loader/directory_refresher.go

```diff
@@ -12,8 +19,15 @@ func (d *DirectoryRefresher) WatchDirectory(runtimePath string, appDirPath strin
 }
 
 func (d *DirectoryRefresher) ShouldRefresh(path string, op FileSystemOp) bool {
-	if filepath.Dir(path) == d.currDir &&
-		(op == Write || op == Create || op == Chmod) {
+	var watchOps *map[FileSystemOp]struct{}
```

Comment:
nit: I recognize this may change as a result of the previous comment, but given an uninitialized map is nil, you can achieve the same here without needing the `*`. Doing so lets you drop the `&` and `*`s in the next few lines.